torch/__init__.py

```diff
@@ -472,6 +472,24 @@ def is_warn_always_enabled():
     """
     return _C._get_warnAlways()
 
+
+def factory_kwargs(kwargs):
+    simple_keys = {"device", "dtype", "memory_format"}
+    expected_keys = simple_keys | {"factory_kwargs"}
+    if not kwargs.keys() <= expected_keys:
```

Comment:
The idea here is that this cannot be used to filter out user-defined arguments and this is only used to unpack the `factory_kwargs`?
Should we rename it to `unpack_factory_kwargs()` ?

Reply:
Yeah, the intention is, "you've taken out all the other kwargs, now assert that it's only factory kwargs left and also do `factory_kwargs` normalization." I don't really want "unpack" in the name because in the standard case no unpacking actually happens (e.g., `{"dtype": "float"}` stays `{"dtype": "float"}`